core/server/api/shared/headers.js

```diff
@@ -37,22 +37,75 @@ const disposition = {
             'Content-Type': 'application/yaml',
             'Content-Length': Buffer.byteLength(JSON.stringify(result))
         };
+    },
+
+    /**
+     * ### Content Disposition Header
+     * create a header that invokes the 'Save As' dialog in the browser when exporting the database to file. The 'filename'
+     * parameter is governed by [RFC6266](http://tools.ietf.org/html/rfc6266#section-4.3).
+     *
+     * For encoding whitespace and non-ISO-8859-1 characters, you MUST use the "filename*=" attribute, NOT "filename=".
+     * Ideally, both. Examples: http://tools.ietf.org/html/rfc6266#section-5
+     *
+     * We'll use ISO-8859-1 characters here to keep it simple.
+     *
+     * @see http://tools.ietf.org/html/rfc598
+     */
+    file(result, options = {}) {
+        return Promise.resolve()
+            .then(() => {
+                let value = options.value;
+
+                if (typeof options.value === 'function') {
+                    value = options.value();
+                }
+
+                return value;
+            })
+            .then((filename) => {
+                return {
+                    'Content-Disposition': `Attachment; filename="${filename}"`
+                };
+            });
     }
 };
 
 module.exports = {
     get(result, apiConfig = {}) {
         let headers = {};
 
-        if (apiConfig.disposition) {
-            Object.assign(headers, disposition[apiConfig.disposition.type](result, apiConfig.disposition));
-        }
+        return Promise.resolve()
```

Comment:
This set of changes could go into master separately as 'Async support for headers' . Lmk if should create another PR for it :)

Reply:
I'd suggest to extract the header async refactoring to master straight? But maybe next Monday? Need to start with QA for tomorrow's release :)

Reply:
sounds good :+1: I'm happy with current state of this PR as is, finally see the end of this over month-long battle to migrate a single controller :laughing:

Reply:
🤫🤣